python/ray/util/client/common.py

```diff
@@ -473,14 +473,23 @@ def _actor_id(self) -> ClientActorRef:
         return self.actor_ref
 
     def __getattr__(self, key):
+        if key == "_method_num_returns":
+            # We need to explicitly handle this value since it is used below,
+            # otherwise we may end up infinitely recursing when deserializing.
+            # This can happen after unpickling an object but before
+            # _method_num_returns is correctly populated.
```

Comment:
ooc, is this tested at all?

Reply:
It's indirectly tested by the deserialization in the unit test, without this part it runs into the recursion limit. But beyond that, no

Reply:
Sounds good--I just wanted to check that it was implicitly tested.